runtime/util/hshelp.c

```diff
@@ -2469,8 +2469,15 @@ recreateRAMClasses(J9VMThread * currentThread, J9HashTable * classHashTable, J9H
 		if (fastHCR) {
 			options |= J9_FINDCLASS_FLAG_FAST_HCR;
 		}
-
-		if (J9_ARE_ALL_BITS_SET(originalRAMClass->classFlags, J9ClassIsAnonymous)) {
+		if (J9_ARE_ANY_BITS_SET(originalRAMClass->romClass->extraModifiers, J9AccClassHiddenClass)) {
+			options |= (J9_FINDCLASS_FLAG_HIDDEN | J9_FINDCLASS_FLAG_UNSAFE | J9_FINDCLASS_FLAG_ANON);
+			if (J9_ARE_ANY_BITS_SET(originalRAMClass->romClass->extraModifiers, J9AccHiddenClassOptionNestmate)) {
```

Comment:
These tests could also be macros in j9modifiers_api.h

Reply:
Added macros.